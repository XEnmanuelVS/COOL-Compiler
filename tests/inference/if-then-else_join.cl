class C {
    
};

class A inherits C {
    
};

class B inherits C {
    
};


class Main {
    func(): AUTO_TYPE {
        if 1 + 1 = 7 then new A else new B fi
    };
};

