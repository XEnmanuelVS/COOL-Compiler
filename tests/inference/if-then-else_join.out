\__ProgramNode [<class> ... <class>]
  \__ClassDeclarationNode: class C  { <feature> ... <feature> }

  \__ClassDeclarationNode: class A : C { <feature> ... <feature> }

  \__ClassDeclarationNode: class B : C { <feature> ... <feature> }

  \__ClassDeclarationNode: class Main  { <feature> ... <feature> }
    \__FuncDeclarationNode: func() : C -> <body>
      \__ConditionalNode: if <expr> then <expr> else <expr> fi
        \__if 
          \__<expr> EqualNode <expr>
            \__<expr> PlusNode <expr>
              \__IntegerNode: 1
              \__IntegerNode: 1
            \__IntegerNode: 7
        \__then 
          \__InstantiateNode: new A()
        \__else 
          \__InstantiateNode: new B()

